src/DataViews/PresetsDataViewTest.cpp

```diff
@@ -0,0 +1,441 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_split.h>
+#include <gmock/gmock-actions.h>
+#include <gmock/gmock-cardinalities.h>
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "DataViews/DataView.h"
+#include "DataViews/PresetLoadState.h"
+#include "DataViews/PresetsDataView.h"
+#include "MetricsUploader/MetricsUploaderStub.h"
+#include "MockAppInterface.h"
+#include "OrbitBase/File.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "OrbitBase/TemporaryFile.h"
+#include "OrbitBase/TestUtils.h"
+#include "PresetFile/PresetFile.h"
+#include "preset.pb.h"
+namespace {
+// This is just a helper type to handle colors. Note that Color from `OrbitGl/CoreMath.h` is not
+// available in this module.
+struct Color {
+  unsigned char r, g, b;
+  friend bool operator==(const Color& lhs, const Color& rhs) {
+    return std::tie(lhs.r, lhs.g, lhs.b) == std::tie(rhs.r, rhs.g, rhs.b);
+  }
+  friend bool operator!=(const Color& lhs, const Color& rhs) { return !(lhs == rhs); }
+};
+}  // namespace
+
+TEST(PresetsDataView, Columns) {
+  orbit_metrics_uploader::MetricsUploaderStub metrics_uploader{};
+  orbit_data_views::MockAppInterface app{};
+  orbit_data_views::PresetsDataView view{&app, &metrics_uploader};
+
+  EXPECT_GE(view.GetColumns().size(), 1);
+  for (const auto& column : view.GetColumns()) {
+    EXPECT_FALSE(column.header.empty());
+  }
+}
+
+TEST(PresetsDataView, Empty) {
+  orbit_metrics_uploader::MetricsUploaderStub metrics_uploader{};
+  orbit_data_views::MockAppInterface app{};
+  orbit_data_views::PresetsDataView view{&app, &metrics_uploader};
+
+  EXPECT_EQ(view.GetNumElements(), 0);
+  EXPECT_EQ(view.GetLabel(), "Presets");
+}
+
+TEST(PresetsDataView, LoadState) {
```

Comment:
CheckLabelAndColorForLoadStates

In general it is better to name test after scenarios rather then members, so that it is clearer what is being tested.

Reply:
Yes. Thanks for suggestion some better names. Is also changed some more.